Cargo.toml

```diff
@@ -26,5 +26,6 @@ linked-hash-map = "0.5.2"
 rand = "0.7"
 rand_distr = "0.2.1"
 test-env-log = "0.2.7"
+insta = "1.8.0"
```

Comment:
FWIW, in rust-analyzer we switched from `insta` to https://github.com/rust-analyzer/expect-test quite a while ago. expect-test is  a bit more minimal. Rather than needing a `cargo-insta` command to upgrade the thest, the test runtime modifies code in-place. 

I kinda like it more, but no strong opinion!

Reply:
That does look reasonably nice! I also don't have a super strong opinion.

Reply:
One nit: you should crate a `cargo expect` command or something instead of using an environment variable! :)

Reply:
Heh, not *having* `cargo expect` is one of the design goals for `expect-test`. I don't like cargo subcommands because:

* they require extra installation steps, you need to teach a human to run `cargo install foo` instead of things just working
* they are not hermetic -- the version of command you use isn't pinned by the Cargo.lockfile, so it can drift over time. 

So that's why expect-test is strictly only a library, but that necessitates somewhat unusual env-var workflow.